src/odemis/driver/smaract.py

```diff
@@ -4043,8 +4282,11 @@ def SA_SI_WaitForEvent(self, handle, event, timeout):
         ev = _deref(event, SA_SI_Event)
         if self.cancel_event:
             self.cancel_event = False
-            ev.error = SA_SI_ERROR_CANCELLED
             raise SA_SIError(SA_SI_ERROR_CANCELLED, "CANCELLED")
+
+        # It's not possible to add ev.error here. Somehow, changing one anonymous parameter of
+        # SA_SI_Event (such as ev.devEventParameter) causes all other anonymous parameters
+        # (ev.error) to store the same value.
```

Comment:
That's because it's an "anonymous" parameter, but a "union". In C, a union means that each member/attribute shares the same memory space. So, of course, changing one changes the other ones. So I don't think this comment is useful here, because it'd be impossible for the C library to change both. Either the error is set with the event type STREAM_ABORTED_EVENT, or a different type is set and the `devEventParameter` is used.